src/main/java/org/sonar/plugins/stash/StashRequestFacade.java

```diff
@@ -183,40 +169,51 @@ void postCommentPerIssue(PullRequestRef pr, String sonarQubeURL,
 
       // Let's call this "issue_loop"
       for (Issue issue : issues) {
-        StashCommentReport comments = commentsByFile.get(issue.getPath());
+        String path = getIssuePath(issue);
+        StashCommentReport comments = commentsByFile.get(path);
+        String commentContent = MarkdownPrinter.printIssueMarkdown(issue, config.getSonarQubeURL());
+        Integer issueLine = issue.line();
+        // FIXME move this somewhere else
+        if (issueLine == null) {
+          issueLine = 0;
+        }
 
         // if comment not already pushed to Stash
-        if ((comments != null) &&
-                (comments.contains(issue.printIssueMarkdown(sonarQubeURL), issue.getPath(), issue.getLine()))) {
-          LOGGER.debug("Comment \"{}\" already pushed on file {} ({})", issue.getKey(),
-                  issue.getPath(), issue.getLine());
+        if (comments != null && comments.contains( commentContent, path, issueLine)) {
+          if (LOGGER.isDebugEnabled()) {
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png 'Severity: CRITICAL') Refactor this code to not nest more than 3 if/for/while/switch/try statements. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS134)